core: leaking connection after 500 from editoast

### What happened?

When for some reasons editoast return 500 like this:
```
[07:50:42,053] [ERROR]     [TimetableProvider] Error when getting http://osrd-rec-editoast/timetable/2998/requirements/?infra_id=361&page=17&page_size=100: Response{protocol=http/1.1, code=500, message=Internal Server Error, url=http://osrd-rec-editoast/timetable/2998/requirements/?infra_id=361&page=17&page_size=100}
```
Logs after indicates that connection was not properly closed
```
 WARNING: A connection to http://osrd-rec-editoast/ was leaked. Did you forget to close a response body? To see where this was allocated, set the OkHttpClient logger level to FINE: Logger.getLogger(OkHttpClient.class.getName()).setLevel(Level.FINE);
Apr 03, 2026 7:50:44 AM okhttp3.internal.platform.Platform log 
```
### What did you expect to happen?

_No response_

### How can we reproduce it (as minimally and precisely as possible)?

Launch a LMR with failing editoast

### On which environments the bug occurs?

Dev (SNCF), Recette (SNCF)

### On which browser the bug occurs?

Firefox

### OSRD version (top right corner `Account` button > `Informations`)

staging 5ce7f87

--- a/core/src/main/java/fr/sncf/osrd/api/TimetableProvider.kt
+++ b/core/src/main/java/fr/sncf/osrd/api/TimetableProvider.kt
@@ -125,6 +125,7 @@ class TimetableDownloader(
                     return response
                 } else {
                     logger.error("Error when getting ${request.url}: $response")
+                    response.close()
                 }
             } catch (e: IOException) {
                 // This block is especially important for timeout errors, but we can retry after any
@@ -137,7 +138,7 @@ class TimetableDownloader(
             // don't want the next page to immediately take over while this one is waiting.
             Thread.sleep(nextSleepDuration)
         }
-        throw UnexpectedHttpResponse(response)
+        throw RuntimeException("Can't access ${request.url}, retry count exceeded")
     }
 
     private data class PaginatedRequirements(

